tests/maintainers.nix

```diff
@@ -0,0 +1,21 @@
+{
+  pkgs ? import <nixpkgs> { },
+  lib ? pkgs.lib,
+}:
+let
+  inherit (lib) attrNames filter length;
+  nixvimList = import ../lib/maintainers.nix;
+  nixpkgsList = lib.maintainers;
+  duplicates = filter (name: nixpkgsList ? ${name}) (attrNames nixvimList);
+  count = length duplicates;
+in
+pkgs.runCommand "maintainers-test" { inherit count duplicates; } ''
+  if [ $count -gt 0 ]; then
+    echo "$count nixvim maintainers are also nixpkgs maintainers:"
+    for name in $duplicates; do
+      echo "- $name"
+    done
+    exit 1
+  fi
+  touch $out
+''
```

Comment:
Isn't this possible using a pure nix check ?
Something like an assertion ?

Reply:
I don't know, couldn't it also support a "classic test". I think nixpkgs has a notion of tests...
Anyway, we could still use `assert` which would cause a dumb derivation to fail.

I just feel like this bash snippet is a bit overkilled.

Reply:
We do have some Nix tests in lib-tests

Reply:
> We do have some Nix tests in lib-tests

Would you rather this be done in `lib-tests`? I felt it made more sense to keep it separate.

Keeping it separate allows:
- Better fail messages
- Future expansion (e.g. check sorting)

Reply:
For the record, lib-tests does essentially the same thing:

https://github.com/nix-community/nixvim/blob/abc409c16fd029f673de194c660df083ba727bd7/tests/lib-tests.nix#L309-L325